vertx-grpc-transcoding/src/main/java/io/vertx/grpc/transcoding/ServiceTranscodingOptions.java

```diff
@@ -0,0 +1,132 @@
+package io.vertx.grpc.transcoding;
+
+import io.vertx.codegen.annotations.VertxGen;
+import io.vertx.core.http.HttpMethod;
+
+import java.util.List;
+
+/**
+ * Defines configuration options for transcoding HTTP requests to gRPC service calls. This interface provides the necessary mapping information to translate between HTTP and gRPC,
+ * including path templates, HTTP methods, body mappings, and additional bindings.
+ *
+ * The transcoding options define how incoming HTTP requests should be mapped to gRPC method calls, including:
+ * <ul>
+ *   <li>Which gRPC service/method to call (selector)</li>
+ *   <li>What HTTP method and path pattern to match</li>
+ *   <li>How to map the HTTP request/response bodies to gRPC messages</li>
+ * </ul>
+ */
+@VertxGen
```

Comment:
This should be `@DataObject` instead in addition of its components.

We should also make those interfaces plain classes as they are rather data containers and not asynchronous and annotation all of them with `@DataObject`